### PR TITLE
Beginnings of a docker-compose production delpoy config

### PR DESCRIPTION
**Pull request by mattkernow**

### CODE REVIEW

File: mapproxy/Dockerfile [hunk 1]
```diff
@@ -21,15 +20,15 @@ RUN apt-get install -y \
...
     libjpeg-dev \
     zlib1g-dev \
     libfreetype6-dev \
     python-virtualenv
 RUN pip install --upgrade pip
-RUN pip install Shapely Pillow MapProxy==1.8.2 uwsgi
+RUN pip install Shapely Pillow MapProxy==1.8.2 uwsgi requests
 
-EXPOSE 8080
+EXPOSE 8080 1717
```
Comment (rightway1) [inline mapproxy/Dockerfile]: This is also in the docker-compose file.  Might avoid confusion to have in one or the other?

File: mapproxy/run.sh [hunk 2]
```diff
@@ -3,6 +3,6 @@ docker kill mapproxy
 docker rm mapproxy
 mkdir mapproxy
 chmod a+wX mapproxy
-docker run --name="mapproxy" -v `pwd`/mapproxy:/mapproxy -p 8080:8080 -d -t thinkwhere/mapproxy
+docker run --name="mapproxy" -v `pwd`home/projects/mapproxy/mapproxy-config:/mapproxy -p 8080:8080 -d -t thinkwhere/mapproxy
```
Comment (rightway1) [inline mapproxy/run.sh]: Lines 3-5 create a folder 'mapproxy' to hold config & cache, but this line is using home/projects/mapproxy/mapproxy-config instead.  Is this just for testing?

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+version: '2'
```
Comment (rightway1) [inline docker-compose.yml:1]: I guess we'll have to decide whether it's best to deploy the docker source to the server from github and build locally, or only deploy the docker-compose and configs, referencing pre-built images from dockerhub.

File: mapproxy/uwsgi.ini [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+#socket=unix:/tmp/mapproxy-uwsgi.socket
+#die-on-term=true
+
+## Enable for testing without NGINX
+#protocol=http
+
+## TODO FOR PRODUCTION!!!
+## THESE SETTINGS ALLOW RUNNING UWSGI AS NON ROOT USER
+#uid=???
+#gid=???
```
Comment (rightway1) [inline mapproxy/uwsgi.ini]: nice work getting all this figured out!

**Comment by mattkernow** [comment]
@rightway1 changes made, back to you

**Pull request merged**